chart/Graphics/Rendering/Chart/Axis/Types.hs

```diff
@@ -367,21 +367,32 @@ makeAxis labelf (labelvs, tickvs, gridvs) = AxisData {
     newViewport = vmap (min',max')
     newTropweiv = invmap (min',max')
     newTicks    = [ (v,2)        | v <- tickvs  ] ++ [ (v,5) | v <- labelvs ]
-    newLabels   = [ (v,labelf v) | v <- labelvs ]
+    newLabels   = zipWithLengthCheck labelvs (labelf labelvs)
+      where
+        zipWithLengthCheck (x:xs) (y:ys) = (x,y) : zipWithLengthCheck xs ys
```

Comment:
I know we mentioned this... But now I think about it... Do you think zipWithLengthCheck is really necessary? A plain old `zip` would drop additional labels, and would only draw those returned, which seems like reasonable behaviour, perhaps better than crashing?